modules/apps/account/account-service/src/main/java/com/liferay/account/internal/configuration/persistence/listener/AccountEntryGroupConfigurationModelListener.java

```
@@ -0,0 +1,79 @@
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.account.internal.configuration.persistence.listener;
+
+import com.liferay.account.constants.AccountConstants;
+import com.liferay.account.internal.configuration.AccountEntryGroupConfiguration;
+import com.liferay.account.internal.settings.AccountEntryGroupSettingsImpl;
+import com.liferay.portal.configuration.persistence.listener.ConfigurationModelListener;
+import com.liferay.portal.configuration.persistence.listener.ConfigurationModelListenerException;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.ListUtil;
+
+import java.util.ArrayList;
+import java.util.Dictionary;
+import java.util.List;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Drew Brokke
+ */
+@Component(
+	immediate = true,
+	property = "model.class.name=com.liferay.account.internal.configuration.AccountEntryGroupConfiguration",
+	service = ConfigurationModelListener.class
+)
+public class AccountEntryGroupConfigurationModelListener
+	implements ConfigurationModelListener {
+
+	@Override
+	public void onBeforeSave(String pid, Dictionary<String, Object> properties)
+		throws ConfigurationModelListenerException {
+
+		String[] allowedTypes = GetterUtil.getStringValues(
+			properties.get("allowedTypes"));
+
+		if (allowedTypes.length == 0) {
+			throw new ConfigurationModelListenerException(
+				"A group must allow at least one account type",
+				AccountEntryGroupConfiguration.class,
+				AccountEntryGroupSettingsImpl.class, properties);
+		}
+
+		List<String> invalidAllowedTypes = new ArrayList<>();
+
+		for (String allowedType : allowedTypes) {
+			if (!ArrayUtil.contains(
+					AccountConstants.ACCOUNT_ENTRY_TYPES_DEFAULT_ALLOWED_TYPES,
+					allowedType)) {
+
+				invalidAllowedTypes.add(allowedType);
```

Can we just throw the exception here when the type is not a valid allowed type?

---

I wanted to provide feedback on every invalid type in case more than one was passed into the `setAllowedTypes ` method.